### PR TITLE
Fix missing custom cypher on unions

### PR DESCRIPTION
**Pull request by angrykoala**
# Description
Add missing Custom Cypher on unions, causing an invalid cypher error

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
Made some changes to the Neo4j GraphQL Toolbox? Or made some changes to the core library perhaps?

Check out your changes at https://neo4j-graphql-toolbox-4218.surge.sh

Use for instance a local dbms hosted on [Neo4j Desktop](https://neo4j.com/download/) or a cloud-hosted [Aura Free](https://neo4j.com/cloud/aura-free/) dbms.

**Comment by neo4j-team-graphql** [comment]
# Performance Report

No Performance Changes


<details>
    <summary>Show Full Table</summary>
    
| name | dbHits | old dbHits | time (ms) | old time (ms) | maxRows |
| ---- | ------ | ---------- | --------- | ------------- | ------- |
|  aggregations.TopLevelAggregate | 3403 | 3403 | 53 | 52 | 1134 |
|  aggregations.NestedAggregation | 16553 | 16553 | 92 | 106 | 2174 |
|  aggregations.AggregationWithWhere | 11979 | 11979 | 59 | 71 | 2174 |
|  aggregations.AggregationWhereWithinNestedRelationships | 22116987 | 22116987 | 4204 | 4287 | 2008534 |
|  aggregations.AggregationWhereWithinNestedConnections | 22116987 | 22116987 | 3753 | 3866 | 2008534 |
|  aggregations.NestedCountFromMovieToActors | 9734 | 9734 | 137 | 59 | 2174 |
|  aggregations.NestedCountFromActorsToMovie | 9937 | 9937 | 81 | 55 | 2174 |
|  aggregations.DeeplyNestedCount | 11062443 | 11062443 | 5921 | 5811 | 2008534 |
|  batch-create.BatchCreate | 4200 | 4200 | 222 | 208 | 600 |
|  batch-create.BatchCreateSmall | 77 | 77 | 119 | 93 | 11 |
|  connect.createAndConnect | 14424 | 14424 | 281 | 305 | 3003 |
|  connections.Connection | 14082 | 14082 | 65 | 86 | 2174 |
|  connections.NestedConnection | 41118 | 41118 | 125 | 171 | 4516 |
|  create.SimpleMutation | 7 | 7 | 80 | 76 | 1 |
|  cypher-directive.TopLevelMutationDirective | 1135 | 1135 | 103 | 94 | 1134 |
|  delete.SimpleDelete | 19401 | 19401 | 292 | 352 | 1040 |
|  delete.NestedDeleteInUpdate | 19198 | 19198 | 315 | 325 | 1179 |
|  2871.NestedRelationshipFilter | 23150 | 23142 | 161 | 102 | 4439 |
|  2925.SingleRelationshipFilter | 6468 | 6468 | 72 | 62 | 1040 |
|  2925.NestedSingleRelationshipFilter | 22900 | 22900 | 104 | 148 | 2174 |
|  2925.SingleRelationshipRequiredFilter | 5201 | 5201 | 53 | 66 | 1040 |
|  2925.NestedSingleRelationshipRequiredFilter | 9361 | 9361 | 72 | 103 | 1040 |
|  query.SimpleQuery | 3121 | 3121 | 27 | 27 | 1040 |
|  query.SimpleQueryWithRelationship | 16162 | 16162 | 57 | 62 | 2174 |
|  query.QueryWhere | 9713 | 9704 | 46 | 50 | 2167 |
|  query.SimpleQueryWithNestedWhere | 9891 | 9882 | 64 | 71 | 2167 |
|  query.Nested | 10096041 | 10096041 | 10377 | 10536 | 2008534 |
|  query.NestedWithFilter | 10074401 | 10074401 | 10648 | 10793 | 2004000 |
|  query.OrFilterOnRelationships | 41175 | 42883 | 214 | 298 | 1360 |
|  query.OrFilterOnRelationshipsAndNested | 35523 | 35761 | 347 | 407 | 1360 |
|  query.QueryWithNestedIn | 13745 | 14262 | 80 | 114 | 1330 |
|  query.NestedConnectionWhere | 9834 | 9834 | 100 | 96 | 2174 |
|  query.DeeplyNestedConnectionWhere | 9898 | 10019 | 115 | 164 | 2174 |
|  query.DeeplyNestedWithRelationshipFilters | 19116 | 19296 | 181 | 244 | 1586 |
|  query.NestedWithRelationshipSingleFilters | 3881 | 3881 | 238 | 365 | 1134 |
|  query.Fulltext | 80 | 80 | 41 | 45 | 16 |
|  query.FulltextWithNestedQuery | 587 | 587 | 52 | 59 | 84 |
|  sorting-and-cypher.TopLevelSortWithCypher | 14092 | 14092 | 62 | 75 | 2174 |
|  sorting-and-cypher.TopLevelConnectionSortWithCypher | 14092 | 14092 | 89 | 97 | 2174 |
|  sorting-and-cypher.TopLevelSortWithCypherWithNested | 14204 | 14205 | 79 | 88 | 2174 |
|  sorting-and-cypher.TopLevelConnectionSortWithCypherWithNested | 14204 | 14205 | 106 | 120 | 2174 |
|  sorting-and-cypher.TopLevelSortWithExpensiveCypher | 14621 | 14711 | 137 | 199 | 2174 |
|  sorting-and-cypher.TopLevelConnectionSortWithExpensiveCypher | 14545 | 14569 | 136 | 184 | 2174 |
|  sorting.SortMultipleTypes | 3515 | 3515 | 96 | 157 | 1040 |
|  sorting.SortMultipleTypesWithCypherWithCypher | 14566 | 14573 | 144 | 279 | 2174 |
|  sorting.SortOnNestedFields | 14082 | 14082 | 99 | 91 | 2174 |
|  sorting.SortDeeplyNestedFields | 43198 | 43198 | 130 | 130 | 4516 |
|  sorting.ConnectionWithSort | 3284 | 3284 | 76 | 98 | 1040 |
|  unions.SimpleUnionQuery | 321 | 321 | 75 | 97 | 35 |
|  unions.SimpleUnionQueryWithMissingFields | 293 | 293 | 74 | 86 | 35 |
|  unions.NestedUnion | 410637 | 410637 | 339 | 385 | 33033 |
|  unions.NestedUnionWithMissingFields | 384611 | 384611 | 322 | 353 | 33033 |
|  update.NestedUpdate | 16143 | 16143 | 156 | 182 | 2002 |

</details>
        
Old Schema Generation: 58.725s
Schema Generation: 52.891s
Old Subgraph Schema Generation: 1:17.394 (m:ss.mmm)
Subgraph Schema Generation: 1:14.777 (m:ss.mmm)

File: packages/graphql/src/translate/queryAST/ast/operations/composite/CompositeReadPartial.ts [hunk 1]
```diff
@@ -62,8 +61,16 @@ export class CompositeReadPartial extends ReadOperation {
...
             matchClause.where(wherePredicate);
         }
-        const subqueries = Cypher.concat(...this.getFieldsSubqueries(nestedContext));
-        const sortSubqueries = wrapSubqueriesInCypherCalls(nestedContext, this.sortFields, [targetNode]);
+        // const subqueries = Cypher.concat(...this.getFieldsSubqueries(nestedContext));
+        // const sortSubqueries = wrapSubqueriesInCypherCalls(nestedContext, this.sortFields, [targetNode]);
+
+        const cypherFieldSubqueries = this.getCypherFieldsSubqueries(nestedContext);
+        const subqueries = Cypher.concat(...this.getFieldsSubqueries(nestedContext), ...cypherFieldSubqueries);
+        // const subqueries = Cypher.concat(...this.getFieldsSubqueries(nestedContext));
```
Comment (MacondoExpress) [inline packages/graphql/src/translate/queryAST/ast/operations/composite/CompositeReadPartial.ts:69]:
```suggestion
        const cypherFieldSubqueries = this.getCypherFieldsSubqueries(nestedContext);
        const subqueries = Cypher.concat(...this.getFieldsSubqueries(nestedContext), ...cypherFieldSubqueries);
```

File: packages/graphql/tests/tck/issues/missing-custom-cypher-on-unions.ts [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+        neoSchema = new Neo4jGraphQL({
+            typeDefs,
+        });
+    });
+
+    test("should include checks for auth jwt param is not null", async () => {
+        const query = gql`
+            query browseHierarchicalComponents($hierarchicalRootId: ID!, $choNodeIris: [ID!]!) {
+                hierarchicalComponents(where: { isContained: { iri: $hierarchicalRootId }, iri_IN: $choNodeIris }) {
+                    #...hierarchicalComponentFields
```
Comment (MacondoExpress) [inline packages/graphql/tests/tck/issues/missing-custom-cypher-on-unions.ts:127]:
```suggestion
```

File: packages/graphql/src/translate/queryAST/ast/operations/composite/CompositeReadPartial.ts [hunk 3]
```diff
@@ -62,8 +61,16 @@ export class CompositeReadPartial extends ReadOperation {
...
+        const sortSubqueries = this.sortFields
```
Comment (mjfwebb) [inline packages/graphql/src/translate/queryAST/ast/operations/composite/CompositeReadPartial.ts:70]: There's a lot of leftover comments here. Was that intended? :thinking:

File: packages/graphql/tests/tck/issues/missing-custom-cypher-on-unions.ts [hunk 4]
```diff
@@ -0,0 +1,236 @@
+/*
```
Comment (MacondoExpress) [inline packages/graphql/tests/tck/issues/missing-custom-cypher-on-unions.ts:1]: The name of the test file does not contains the suffix `.test.`

**Review by MacondoExpress — COMMENTED** [review]
Left some comments, also it's missing the changeset!

File: packages/graphql/src/translate/queryAST/ast/operations/composite/CompositeReadPartial.ts (hunk 3 above)
Comment (angrykoala) [reply packages/graphql/src/translate/queryAST/ast/operations/composite/CompositeReadPartial.ts:70]: Nope, sorry I missed those

File: packages/graphql/tests/tck/issues/missing-custom-cypher-on-unions.ts (hunk 2 above)
Comment (angrykoala) [reply packages/graphql/tests/tck/issues/missing-custom-cypher-on-unions.ts:127]: This is actually part of the initial bug report, so I think we should leave it

File: packages/graphql/tests/tck/issues/missing-custom-cypher-on-unions.ts (hunk 4 above)
Comment (angrykoala) [reply packages/graphql/tests/tck/issues/missing-custom-cypher-on-unions.ts]: Good catch

File: packages/graphql/tests/tck/issues/missing-custom-cypher-on-unions.ts (hunk 2 above)
Comment (MacondoExpress) [reply packages/graphql/tests/tck/issues/missing-custom-cypher-on-unions.ts]: Sounds good!

**Pull request merged**